### PR TITLE
Fix Linux build 

### PR DESCRIPTION
**Pull request by Nicky-D**
Before the Linux build was not properly OpenGL 4.5 aware. Fix this by making sure the build VM has all the required dev packages.

@nat-goodspeed I have to shine your call sign over Linden Lab HQ again, you are the hero we need

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -15,12 +15,31 @@ jobs:
...
         if: ${{ runner.os == 'Linux' }}
         run: sudo apt-get install ninja-build
+
+      - name: Run apt update
+        if: runner.os == 'Linux'
+        run: sudo apt update
+
+      - name: Run apt update
+        if: runner.os == 'Linux'
+        run: sudo apt upgrade
```
Comment (bennettgoble) [inline .github/workflows/build.yaml]: Is the system-wide package update needed? We have generally been leaving updating system packages to the source VM images and only pulling in updated dependencies specific to the build. (Follows [GH's customization instructions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/customizing-github-hosted-runners)) Running a full upgrade also takes more time.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -15,12 +15,31 @@ jobs:
       - name: Setup Linux Build Dependencies
         if: ${{ runner.os == 'Linux' }}
         run: sudo apt-get install ninja-build
+
+      - name: Run apt update
+        if: runner.os == 'Linux'
+        run: sudo apt update
```
Comment (bennettgoble) [inline .github/workflows/build.yaml:17]: Nit: combine this with the package install into a single step. Less steps, less conditions.

**Review by bennettgoble — CHANGES_REQUESTED** [review]
Thanks for the PR! I love seeing linux builds getting added. :) I had a few suggestions, mostly nit-picky.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (Nicky-D) [reply .github/workflows/build.yaml]: I'm not 100% sure as I've not tried. I can try later if it works and produces a SDL2 library the viewer can consume and create an OpenGL context with

**Review by nat-goodspeed — APPROVED** [review]
I agree that folding all the `runner.os == 'Linux'` steps into a single multi-line step would make more sense. That said, I'll try building a release as-is.

**Comment by nat-goodspeed** [comment]
Your PR is on our `linux` branch, not yet `main`, but here's the build:
https://github.com/secondlife/3p-sdl2/releases/tag/v2.28.0-1dc88c1

**Review by bennettgoble — APPROVED** [review]
Kicking off a build, the changes look good to me --thanks for the updates, @Nicky-D. I'll get a pre-release published if the CI passes.

**Comment by Nicky-D** [comment]
> Kicking off a build, the changes look good to me --thanks for the updates, @Nicky-D. I'll get a pre-release published if the CI passes.

Seems like without apt update at least it is not passing on LL's runner (or some standard GitHub runners are different?) So I added that back

**Review by bennettgoble — APPROVED** [review]

**Comment by bennettgoble** [comment]
@Nicky-D thanks for the fixups. The new release is https://github.com/secondlife/3p-sdl2/releases/tag/v2.28.0-r2

**Pull request merged**